stake-pool/cli/src/main.rs

```diff
@@ -391,12 +391,80 @@ fn pool_tokens_to_stake_amount(pool_data: &StakePool, tokens: u64) -> u64 {
         .unwrap() as u64
 }
 
+struct WithdrawAccount {
+    pubkey: Pubkey,
+    account: Account,
+    amount: u64,
+}
+
+fn prepare_withdraw_accounts(
+    config: &Config,
+    pool_withdraw_authority: &Pubkey,
+    amount: u64,
+) -> Result<Vec<WithdrawAccount>, Error> {
+    let mut accounts = get_authority_accounts(config, &pool_withdraw_authority);
+    if accounts.is_empty() {
+        return Err("No accounts found.".to_string().into());
+    }
+    // Sort from lowest to highest balance
+    accounts.sort_by(|a, b| a.1.lamports.cmp(&b.1.lamports));
+
+    // Prepare the list of accounts to withdraw from
+    let mut withdraw_from: Vec<WithdrawAccount> = vec![];
+    let mut remaining_amount = amount;
+
+    // First use largest amounts
+    while !accounts.is_empty() && accounts.last().unwrap().1.lamports < remaining_amount {
+        let (pubkey, account) = accounts.last().unwrap();
+
+        // Those accounts will be withdrawn completely with `claim` instruction
+        withdraw_from.push(WithdrawAccount {
+            pubkey: *pubkey,
+            account: account.clone(),
+            amount: account.lamports,
+        });
+        remaining_amount -= account.lamports;
+        accounts.pop();
+    }
+    if remaining_amount > 0 {
```

Comment:
Won't this always be true?

Reply:
Yes, the error was `<` instead of `<=` in the condition above, then it could be zero. As per your next comment I'll better join two loops together.

Reply:
This one it outdated now, will describe what was done in the next comment.